superset-frontend/src/explore/components/SaveModal.tsx

```diff
@@ -150,9 +151,6 @@ class SaveModal extends React.Component<SaveModalProps, SaveModalState> {
   async saveOrOverwrite(gotodash: boolean) {
     this.setState({ alert: null, isLoading: true });
     this.props.actions.removeSaveModalAlert();
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-
-    let promise = Promise.resolve();
```

Comment:
Changing promises to `await`s looks great, thanks!  Two related notes:
1. Since all the `catch` blocks are the same, should we just wrap the entire function body in one `try` block?
2. Previously, since these promises had no `catch` block, errors would go unhandled and therefore show up in the browser console.  Should we preserve that behavior by re-throwing or not catching errors?

If you think yes for both of those, we could just do something like:
```javascript
async saveOrOverwrite(goodish:boolean) {
  try {
    // If anything throws here, modal will stay open, finally block will execute,
    // and error will remain unhandled because there's no catch block
    this.props.onHide();
  } finally {
    this.setState({ isLoading: false });
  }
}

Reply:
Great idea! Done